cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -755,17 +775,36 @@ updateTopologyConfiguration startupTracer nc localRootsVar publicRootsVar useLed
                 $ pack "Error reading topology configuration file:" <> err
       Right nt@RealNodeTopology { ntUseLedgerPeers
                                 , ntUseBootstrapPeers
+                                , ntPeerSnapshotPath
                                 } -> do
         let (localRoots, publicRoots) = producerAddresses nt
         traceWith startupTracer
-                $ NetworkConfig localRoots publicRoots ntUseLedgerPeers
+                $ NetworkConfig localRoots publicRoots ntUseLedgerPeers ntPeerSnapshotPath
         atomically $ do
           writeTVar localRootsVar localRoots
           writeTVar publicRootsVar publicRoots
           writeTVar useLedgerVar ntUseLedgerPeers
           writeTVar useBootsrapPeersVar ntUseBootstrapPeers
+          writeTVar ledgerPeerSnapshotPathVar ntPeerSnapshotPath
 #endif
 
+updateLedgerPeerSnapshot :: Tracer IO (StartupTrace blk)
+                         -> STM IO (Maybe PeerSnapshotFile)
+                         -> (Maybe LedgerPeerSnapshot -> STM IO ())
+                         -> IO (Maybe LedgerPeerSnapshot)
+updateLedgerPeerSnapshot startupTracer readLedgerPeerPath writeVar = runMaybeT $
+              (\io_m_lps -> do
+                  m_lps <- io_m_lps
```

Comment:
Why not camelcase?

Reply:
I though io_m_lps was a little clearer than ioMLps, or do you have something else in mind?

Reply:
That's because you're trying to shoehorn your function into `MaybeT` which doesn't work here very well - you're doing most of the stuff in `IO . Maybe` anyway. Why not write this function like this:
```haskell
updateLedgerPeerSnapshot :: Tracer IO (StartupTrace blk)
                         -> STM IO (Maybe PeerSnapshotFile)
                         -> (Maybe LedgerPeerSnapshot -> STM IO ())
                         -> IO (Maybe LedgerPeerSnapshot)
updateLedgerPeerSnapshot startupTracer readLedgerPeerPath writeVar = do
  mPeerSnapshotFile <- atomically readLedgerPeerPath
  mLedgerPeerSnapshot <- forM mPeerSnapshotFile $ \f -> do
    lps@(LedgerPeerSnapshot (wOrigin, _)) <- readPeerSnapshotFile f
    lps <$ traceWith startupTracer (LedgerPeerSnapshotLoaded wOrigin)
  atomically . writeVar $ mLedgerPeerSnapshot
  pure mLedgerPeerSnapshot
```

Reply:
Yes, that's most intuitive.